clean_air/util/cubes.py

```diff
@@ -18,3 +22,107 @@ def get_xy_coords(cube):
     xcoord = cube.coord(axis="x", dim_coords=True)
     ycoord = cube.coord(axis="y", dim_coords=True)
     return xcoord, ycoord
+
+
+def extract_box(cube, box):
+    """
+    Extracts a rectangular area from a cube.
+
+    Args:
+        cube: Cube to subset
+        box (x0, y0, x1, y1): Box defined in terms of its lower-left (x0, y0)
+            and upper-right (x1, y1) corners
+    """
+
+    def extent_checker(low, high):
```

Comment:
I'm finding your coding style very technical here and a little confusing.  I think whenever I've seen or used private functions like this before they've started with an underscore to indicate that they are private helper functions.  I'm not sure if this is a python best practice or just something that they did in AVD, but it did help me to understand that they are meant to be used specifically for this (or whatever) purpose and shouldn't ever be imported and used somewhere else.  Is that what this is?

Reply:
I think there's a few concepts being conflated here.

The convention of private things (functions/constants/whatever) beginning with an underscore is absolutely true, which exists because, as I'm sure you know, nothing in python is "actually" private. It's an assumption python itself makes, for example it will not suggest names beginning with an underscore during tab-completion.

But, this convention doesn't even apply here. The inner function `extent_checker` is not accessible in any way from outside `extract_box` because it literally does not exist. It's a variable that's local to the function, just like all the others - `cube`, `xcoord` and so on - just a variable that happens to be a function.

I realise it looks similar to defining a "private" *method* of a class, where the function would be created at the same time as the class, and is accessible as long as the class is. So I would understand wanting to use underscores anyway, just to avoid readers needing to reassure themselves that yes this not publicly accessible because it's in a `def` not a `class`. Would that help?

Reply:
I think so, if that's a thing that you're happy with.  My mind would be a little more rested if you used the underscores, but if it's not proper then you don't have to.

Reply:
And also thank you for the explanation, that helps a lot.